### PR TITLE
Deviate from JTS in extra-dimension coordinate handling.

### LINKED ISSUE #461: Proposed deviation from JTS on resolution to locationtech/jts#580
@FObermaier

(cc: @dr-jts in case you have any thoughts about this topic)

Given that the only actual **problem** stated in locationtech/jts#580 is that users would lose some of their coordinate data when creating `Coordinate` instances, I propose reverting b713c2b and fixing that issue by removing `PackedCoordinateSequence`'s override of `GetCoordinateCopy` instead.

This is because the default behavior in NTS, when not overridden, will return an instance that contains all ordinate values, even if the user accidentally asked for 4 spatial dimensions.  We have specific tests for this behavior, so we might as well not override it just for the sake of making it only support a narrower set of use cases.  Specific tests are:
https://github.com/NetTopologySuite/NetTopologySuite/blob/17684339365c57f429b173b3d82b2a1ff9d84b78/test/NetTopologySuite.Tests.NUnit/Geometries/Implementation/BaseCoordinateSequenceTest.cs#L255-L273

Given that this would fix the only actual reported **problem** from that issue, I don't think it's right to change the behavior of `CoordinateSeqeuenceFactory.Create(int size, int dimension)`, since this is a break from 2.0 / 2.1...

### PR DESCRIPTION
**Pull request by airbreather**
- 4th spatial dimension is now accessible in all cases.
- Certain coordinate sequences now deviate from JTS to support more cases.

Resolves #461

### CODE REVIEW

File: src/NetTopologySuite/Geometries/Implementation/CoordinateArraySequenceFactory.cs [hunk 1]
```diff
@@ -36,17 +36,6 @@ public override CoordinateSequence Create(int size, int dimension, int measures)
         {
             int spatial = dimension - measures;
 
-            if (measures > 1)
```
Comment (FObermaier) [inline src/NetTopologySuite/Geometries/Implementation/CoordinateArraySequenceFactory.cs:39]: We should note that we deliberatly deviate from JTS' CoordinateArraySequence.

File: src/NetTopologySuite/Geometries/Implementation/PackedCoordinateSequence.cs [hunk 2]
```diff
@@ -43,43 +43,7 @@ public sealed override Coordinate GetCoordinate(int i)
             var arr = GetCachedCoords();
             if(arr != null)
                  return arr[i];
-            return GetCoordinateInternal(i);
```
Comment (FObermaier) [inline src/NetTopologySuite/Geometries/Implementation/PackedCoordinateSequence.cs:46]: We should note that we deliberatly deviate from JTS' PackedCoordinateSequence.

Comment (FObermaier) [reply src/NetTopologySuite/Geometries/Implementation/PackedCoordinateSequence.cs:46]: Perhaps put it in `#region ... end region` and comment the code.

File: src/NetTopologySuite/Geometries/Implementation/PackedCoordinateSequence.cs [hunk 3]
```diff
@@ -132,69 +96,6 @@ private Coordinate[] GetCachedCoords()
             return null;
         }
 
-        /// <summary>
```
Comment (FObermaier) [inline src/NetTopologySuite/Geometries/Implementation/PackedCoordinateSequence.cs:135]: dito

**Review by FObermaier — APPROVED** [review]
Consider my annotations as suggestion.
I think it is easier to maintain this code if we are remembered that for some reason or other we deliberatly deviate from the JTS' original.

**Comment by FObermaier** [comment]
@airbreather, another idea is to place a README.md file into `NetTopologySuite.Geometries.Implementation` stating that the content of the whole namespace is different than `org.locationtech.jts.geom.impl` and that changes over there should only be ported if the related unit test fails on NTS, too.

I'd like to release v2.2 soon, that is why i assigned some issues to milestone 2.2. I'd be grateful if you could have a look at those and tell me if you'd like to see any of these addressed before a new release. Thanks!

**Comment by airbreather** [comment]
Ugh.  I was trying to get the initialization from `Coordinate[]` to consistently take the highest dimension and measures from the given array, but it actually seems like using `CoordinateArrays.Dimension` and `CoordinateArrays.Measures` separately is the wrong thing to do.

What we really need is something that tells us the smallest (dimension, measures) that can hold all the same slots from the given coordinate array.

As-is, a `Coordinate[]` with some `CoordinateZ` and some `CoordinateM` would tell us to use dimension 3 and measures 1, but we really would need dimension 4 and measures 1... I'll roll some of that last commit back and have everything keep looking at just the first coordinate.

**Review by FObermaier — APPROVED** [review]
We should not use the obsolete constructors ourselves. Other than that I'm fine with this.

**Pull request merged**